messages.proto

```diff
@@ -30,32 +30,64 @@ enum MessageRight {
   CAN_DELETE_MESSAGE = 2;
 }
 
+// Full message with all fields
 message Message {
   ChatPreview chat = 8;
   string id = 1;
   uint64 created_at = 3;
-  uint64 updated_at = 9;
+  uint64 changed_at = 9; // when equal to created_at, message was not edited
   string text = 4;
   User sender = 5;
   string prev_message_id = 6;
   uint32 viewed_counter = 7; // how many recipients viewed this message
   repeated MessageRight rights = 10;
+
+  // attachments
+  MessageReference reply = 12;
```

Comment:
What's the difference between forward a single message from the same chat and reply?